compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/hotspot/aarch64/AArch64HotSpotNodeLIRBuilder.java

```diff
@@ -105,7 +105,7 @@ protected void emitPrologue(StructuredGraph graph) {
             assert stub.getLinkage().getDescriptor().getTransition() != HotSpotForeignCallDescriptor.Transition.SAFEPOINT : stub;
             Register[] savedRegisters = getGen().getRegisterConfig().getAllocatableRegisters().toArray();
             AArch64SaveRegistersOp saveOp = getGen().emitSaveAllRegisters(savedRegisters);
-            append(saveOp);
+            assert getGen().lastIs(saveOp);
```

Comment:
not sure the assertion is worth it. either `emitSaveAllRegisters` is supposed to append it (I think it should), or not.

Reply:
It might be worth adding a more general assert in `LIRGenerator.append` like
```
assert !lirForBlock.contains(op) : "added " + op + " twice";
```